docs/src/references/agent/device-management-api.md

```diff
@@ -0,0 +1,295 @@
+```
+
+### Command status topics
+
+The actual command requests are published on the __command status topics__.
+For each request, a specific command topic is created to monitor the progress of the command from its initial state to its completion.
+These topics are named using a unique command identifier forged by the requester.
+
+```mermaid
+graph LR
+  te --/--- identifier --/--- cmd
+  subgraph root
+    te
+  end
+
+  subgraph target
+    identifier["&lt;target identifier&gt;"]
+  end
+
+  subgraph command["command request"]
+    direction LR
+    cmd --/--- cmd_type["&lt;command_type&gt;"] --/--- cmd_id["&lt;cmd_id&gt;"]
+  end
+```
+
+| Group        | Description                               |
+|--------------|-------------------------------------------|
+| command_id   | The identifier of the operation instance. |
+
+The messages published on these topics represent each the current status of a running command.
+So, one can list __all the pending commands of any type across all the devices__:
+
+```sh te2mqtt formats="v1"
+tedge mqtt sub 'te/+/+/+/+/cmd/+/+'
+```
+
+## Message payloads
+
+The message payloads are all specific to each operation type.
+However, there are some common rules:
+
+- JSON is used for all messages.
+  - The schemas are flexible.
+  - Extra fragments can be added by participants and ignored by others if not relevant.
+- Device capabilities are published as retained messages.
+  - If a capability has no specific metadata, then the capability is published as an empty JSON object `{}`.
+  - A capability is removed by publishing a retained empty string `""`.
+- Command status messages are published as retained messages on their dedicated topic.
+  - These messages represent the current state of the command.
+  - Initialized with all the required information to execute the command,
+    they are updated during the command execution with outcome data.
+  - A command request is cleared by publishing a retained empty string `""`.
+- The command status messages contain a mandatory `"status"` field.
+  - This status tells at which step the command execution is along its workflow. 
+  - A new command is created with a `"status"` set to `"init"`.
+  - When execution starts, the `"status"` is set to `"executing"`.
+  - When execution is successful, the `"status"` is set to `"successful"`.
+  - If execution fails for some reason, the `"status"` is set to `"failed"` and the reason given as `"reason"` field.
+  - An operation might define specific `"status"` values for extra steps, actions or checks.
+
+:::info
+Health checks is a notable exception to the rule "command status are published as retained messages".
+A health check is sent to the chanel `cmd/health/check` of a service (or a device) as a non-retained request `{}`,
+triggering a health message response published on the `status/health` channel of the same service
+(or respectively the channels of all the service running on that device).
+:::
```

Comment:
I have a concern about this use of the `cmd` channel for health check.

I would really prefer to send  health check requests on a `req` or `request` channel.
There are indeed too many differences:
- Health check requests must not be retained
- There is no need for a specific id for each request
- The response is sent to another topic (`status/health`).

Having such exceptions will make the implementation of extensible command workflow a bit more complex for no added value.

Having a `req/<request-type>` channel can be used for other kind of requests: [setpoints](https://en.wikipedia.org/wiki/Setpoint_(control_system)), clearing alarms ...

Reply:
I wasn't a big fan of retrofitting health statuses by partially using the cmd APIs. But we did that at that point to avoid introducing yet another req/res mechanism just for health statuses. But, with additional things like setpoints or clear alarm requests, may be it's worth adding something.

But, I still wouldn't define an API that ties the response to the request in any way, as we want components to freely declare health status, setpoint values etc without any explicit request. Even for the clear alarm case, the prompt will be something and the resultant action will be just the regular "clear alarm" message. Will have to think about such a disjoint req/prompt mechanism.

Reply:
For details, see https://github.com/thin-edge/thin-edge.io/issues/2372